File: examples/python/minimal.py
import sys
import os

# Add bindings/python to system path so we can import fluxui
sys.path.append(os.path.join(os.path.dirname(__file__), "../../bindings/python"))

from fluxui import App

def main():
    app = App()
    if not app.init("FluxUI Python Minimal Example", 800, 600):
        print("Failed to initialize FluxUI application.")
        return

    root = app.root()
    
    panel = root.add_panel("container")
    panel.style_width(600)
    panel.style_height(400)
    panel.style_background_color(0.1, 0.1, 0.15, 1.0)
    
    label = panel.add_text("Welcome to FluxUI from Python!")
    
    counter = [0]
    
    def on_click():
        counter[0] += 1
        label.set_content(f"Button Clicked {counter[0]} times!")
        print(f"Clicked: {counter[0]}")
        
    btn = panel.add_button("Click Me")
    btn.set_on_click(on_click)
    
    exit_btn = panel.add_button("Exit")
    exit_btn.set_on_click(lambda: app.stop())

    print("Running FluxUI app...")
    app.run()

if __name__ == "__main__":
    main()

File: examples/go/minimal.go
package main

import (
	"fmt"
	"os"
	"runtime"

	"github.com/camiloGHS21/DataleakGuard/bindings/go"
)

func main() {
	// Lock OS thread because UI frameworks usually require running on the main thread
	runtime.LockOSThread()

	app, err := fluxui.CreateApp()
	if err != nil {
		fmt.Printf("Error creating app: %v\n", err)
		os.Exit(1)
	}
	defer app.Destroy()

	fmt.Printf("App handle: 0x%x\n", app.GetHandle())
	app.SetBackend(100) // 100 is FLUXUI_BACKEND_COMPATIBILITY
	if !app.Init("FluxUI Go Minimal Example", 800, 600) {
		fmt.Println("Failed to initialize FluxUI App")
		os.Exit(1)
	}
	fmt.Printf("Active Backend: %d\n", app.GetBackend())

	root := app.Root()
	fmt.Printf("Root widget handle: 0x%x\n", root.GetHandle())
	panel := root.AddPanel("container")
	fmt.Printf("Panel widget handle: 0x%x\n", panel.GetHandle())
	panel.StyleWidth(600)
	panel.StyleHeight(400)
	panel.StyleBackgroundColor(0.15, 0.1, 0.1, 1.0)

	label := panel.AddText("Welcome to FluxUI from Go!", "")
	_ = label

	counter := 0
	_ = counter
	btn := panel.AddButton("Click Me", "btn")
	_ = btn
	// btn.SetOnClick(func() {
	// 	counter++
	// 	label.SetContent(fmt.Sprintf("Button clicked %d times!", counter))
	// 	fmt.Printf("Clicked: %d\n", counter)
	// })

	exitBtn := panel.AddButton("Exit", "btn")
	_ = exitBtn
	// exitBtn.SetOnClick(func() {
	// 	app.Stop()
	// })

	fmt.Println("Running FluxUI app from Go...")
	app.Run()
}
